File: app/src/main/java/com/example/jaguar/weather/model/network/UserModelResponse.java
package com.example.jaguar.weather.model.network;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import java.util.List;

public class UserModelResponse {
    @SerializedName("location")
    @Expose
    public Location location;
    @SerializedName("current")
    @Expose
    public Current current;
    @SerializedName("forecast")
    @Expose
    public Forecast forecast;

    public class Location {
        public String getName() {
            return name;
        }

        @SerializedName("name")
        @Expose
        public String name;

        public String getRegion() {
            return region;
        }

        @SerializedName("region")
        @Expose
        public String region;
        @SerializedName("country")
        @Expose
        public String country;
        @SerializedName("lat")
        @Expose
        public float lat;
        @SerializedName("lon")
        @Expose
        public float lon;
        @SerializedName("tz_id")
        @Expose
        public String tzId;
        @SerializedName("localtime_epoch")
        @Expose
        public int localtimeEpoch;
        @SerializedName("localtime")
        @Expose
        public String localtime;
    }

    public class Current {
        @SerializedName("last_updated_epoch")
        @Expose
        public int lastUpdatedEpoch;
        @SerializedName("last_updated")
        @Expose
        public String lastUpdated;
        @SerializedName("temp_c")
        @Expose
        public float tempC;
        @SerializedName("temp_f")
        @Expose
        public float tempF;
        @SerializedName("is_day")
        @Expose
        public int isDay;
        @SerializedName("condition")
        @Expose
        public Condition condition;
        @SerializedName("wind_mph")
        @Expose
        public float windMph;
        @SerializedName("wind_kph")
        @Expose
        public float windKph;
        @SerializedName("wind_degree")
        @Expose
        public int windDegree;
        @SerializedName("wind_dir")
        @Expose
        public String windDir;
        @SerializedName("pressure_mb")
        @Expose
        public float pressureMb;
        @SerializedName("pressure_in")
        @Expose
        public float pressureIn;
        @SerializedName("precip_mm")
        @Expose
        public float precipMm;
        @SerializedName("precip_in")
        @Expose
        public float precipIn;
        @SerializedName("humidity")
        @Expose
        public int humidity;
        @SerializedName("cloud")
        @Expose
        public int cloud;
        @SerializedName("feelslike_c")
        @Expose
        public float feelslikeC;
        @SerializedName("feelslike_f")
        @Expose
        public float feelslikeF;
        @SerializedName("vis_km")
        @Expose
        public float visKm;
        @SerializedName("vis_miles")
        @Expose
        public float visMiles;
        @SerializedName("uv")
        @Expose
        public float uv;
        @SerializedName("gust_mph")
        @Expose
        public float gustMph;
        @SerializedName("gust_kph")
        @Expose
        public float gustKph;

        public class Condition {
            @SerializedName("text")
            @Expose
            public String text;

            public String getIcon() {
                return icon;
            }

            @SerializedName("icon")
            @Expose
            public String icon;
            @SerializedName("code")
            @Expose
            public int code;

        }
    }
    public class Forecast {
        @SerializedName("forecastday")
        @Expose
        public List<Forecastday> forecastday = null;
        public class Forecastday {
            @SerializedName("date")
            @Expose
            public String date;
            @SerializedName("date_epoch")
            @Expose
            public int dateEpoch;
            @SerializedName("day")
            @Expose
            public Day day;
            @SerializedName("astro")
            @Expose
            public Astro astro;
            public class Day {
                @SerializedName("maxtemp_c")
                @Expose
                public float maxtempC;
                @SerializedName("maxtemp_f")
                @Expose
                public float maxtempF;
                @SerializedName("mintemp_c")
                @Expose
                public float mintempC;
                @SerializedName("mintemp_f")
                @Expose
                public float mintempF;
                @SerializedName("avgtemp_c")
                @Expose
                public float avgtempC;
                @SerializedName("avgtemp_f")
                @Expose
                public float avgtempF;
                @SerializedName("maxwind_mph")
                @Expose
                public float maxwindMph;
                @SerializedName("maxwind_kph")
                @Expose
                public float maxwindKph;
                @SerializedName("totalprecip_mm")
                @Expose
                public float totalprecipMm;
                @SerializedName("totalprecip_in")
                @Expose
                public float totalprecipIn;
                @SerializedName("avgvis_km")
                @Expose
                public float avgvisKm;
                @SerializedName("avgvis_miles")
                @Expose
                public float avgvisMiles;
                @SerializedName("avghumidity")
                @Expose
                public float avghumidity;
                @SerializedName("condition")
                @Expose
                public Condition_ condition;
                @SerializedName("uv")
                @Expose
                public float uv;
            }
            public class Condition_ {
                @SerializedName("text")
                @Expose
                public String text;
                @SerializedName("icon")
                @Expose
                public String icon;
                @SerializedName("code")
                @Expose
                public int code;
            }

            public class Astro {
                @SerializedName("sunrise")
                @Expose
                public String sunrise;
                @SerializedName("sunset")
                @Expose
                public String sunset;
                @SerializedName("moonrise")
                @Expose
                public String moonrise;
                @SerializedName("moonset")
                @Expose
                public String moonset;

            }
        }
    }
}

File: app/src/main/java/com/example/jaguar/weather/common/UserModelResponse.kt
package com.example.jaguar.weather.common

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class UserModelResponse {
    @SerializedName("location")
    @Expose
    lateinit var location: Location
    @SerializedName("current")
    @Expose
    lateinit var current: Current
    @SerializedName("forecast")
    @Expose
    lateinit var forecast: Forecast

    inner class Location {

        @SerializedName("name")
        @Expose
        lateinit var name: String

        @SerializedName("region")
        @Expose
        lateinit var region: String
        @SerializedName("country")
        @Expose
        lateinit var country: String
        @SerializedName("lat")
        @Expose
        var lat: Float = 0.toFloat()
        @SerializedName("lon")
        @Expose
        var lon: Float = 0.toFloat()
        @SerializedName("tz_id")
        @Expose
        lateinit var tzId: String
        @SerializedName("localtime_epoch")
        @Expose
        var localtimeEpoch: Int = 0
        @SerializedName("localtime")
        @Expose
        lateinit var localtime: String
    }

    inner class Current {
        @SerializedName("last_updated_epoch")
        @Expose
        var lastUpdatedEpoch: Int = 0
        @SerializedName("last_updated")
        @Expose
        lateinit var lastUpdated: String
        @SerializedName("temp_c")
        @Expose
        var tempC: Float = 0.toFloat()
        @SerializedName("temp_f")
        @Expose
        var tempF: Float = 0.toFloat()
        @SerializedName("is_day")
        @Expose
        var isDay: Int = 0
        @SerializedName("condition")
        @Expose
        lateinit var condition: Condition
        @SerializedName("wind_mph")
        @Expose
        var windMph: Float = 0.toFloat()
        @SerializedName("wind_kph")
        @Expose
        var windKph: Float = 0.toFloat()
        @SerializedName("wind_degree")
        @Expose
        var windDegree: Int = 0
        @SerializedName("wind_dir")
        @Expose
        lateinit var windDir: String
        @SerializedName("pressure_mb")
        @Expose
        var pressureMb: Float = 0.toFloat()
        @SerializedName("pressure_in")
        @Expose
        var pressureIn: Float = 0.toFloat()
        @SerializedName("precip_mm")
        @Expose
        var precipMm: Float = 0.toFloat()
        @SerializedName("precip_in")
        @Expose
        var precipIn: Float = 0.toFloat()
        @SerializedName("humidity")
        @Expose
        var humidity: Int = 0
        @SerializedName("cloud")
        @Expose
        var cloud: Int = 0
        @SerializedName("feelslike_c")
        @Expose
        var feelslikeC: Float = 0.toFloat()
        @SerializedName("feelslike_f")
        @Expose
        var feelslikeF: Float = 0.toFloat()
        @SerializedName("vis_km")
        @Expose
        var visKm: Float = 0.toFloat()
        @SerializedName("vis_miles")
        @Expose
        var visMiles: Float = 0.toFloat()
        @SerializedName("uv")
        @Expose
        var uv: Float = 0.toFloat()
        @SerializedName("gust_mph")
        @Expose
        var gustMph: Float = 0.toFloat()
        @SerializedName("gust_kph")
        @Expose
        var gustKph: Float = 0.toFloat()

        inner class Condition {
            @SerializedName("text")
            @Expose
            lateinit var text: String

            @SerializedName("icon")
            @Expose
            lateinit var icon: String
            @SerializedName("code")
            @Expose
            var code: Int = 0

        }
    }

    inner class Forecast {
        @SerializedName("forecastday")
        @Expose
        lateinit var forecastday: List<Forecastday>

        inner class Forecastday {
            @SerializedName("date")
            @Expose
            lateinit var date: String
            @SerializedName("date_epoch")
            @Expose
            var dateEpoch: Int = 0
            @SerializedName("day")
            @Expose
            lateinit var day: Day
            @SerializedName("astro")
            @Expose
            lateinit var astro: Astro

            inner class Day {
                @SerializedName("maxtemp_c")
                @Expose
                var maxtempC: Float = 0.toFloat()
                @SerializedName("maxtemp_f")
                @Expose
                var maxtempF: Float = 0.toFloat()
                @SerializedName("mintemp_c")
                @Expose
                var mintempC: Float = 0.toFloat()
                @SerializedName("mintemp_f")
                @Expose
                var mintempF: Float = 0.toFloat()
                @SerializedName("avgtemp_c")
                @Expose
                var avgtempC: Float = 0.toFloat()
                @SerializedName("avgtemp_f")
                @Expose
                var avgtempF: Float = 0.toFloat()
                @SerializedName("maxwind_mph")
                @Expose
                var maxwindMph: Float = 0.toFloat()
                @SerializedName("maxwind_kph")
                @Expose
                var maxwindKph: Float = 0.toFloat()
                @SerializedName("totalprecip_mm")
                @Expose
                var totalprecipMm: Float = 0.toFloat()
                @SerializedName("totalprecip_in")
                @Expose
                var totalprecipIn: Float = 0.toFloat()
                @SerializedName("avgvis_km")
                @Expose
                var avgvisKm: Float = 0.toFloat()
                @SerializedName("avgvis_miles")
                @Expose
                var avgvisMiles: Float = 0.toFloat()
                @SerializedName("avghumidity")
                @Expose
                var avghumidity: Float = 0.toFloat()
                @SerializedName("condition")
                @Expose
                lateinit var condition: Condition_
                @SerializedName("uv")
                @Expose
                var uv: Float = 0.toFloat()
            }

            inner class Condition_ {
                @SerializedName("text")
                @Expose
                lateinit var text: String
                @SerializedName("icon")
                @Expose
                lateinit var icon: String
                @SerializedName("code")
                @Expose
                var code: Int = 0
            }

            inner class Astro {
                @SerializedName("sunrise")
                @Expose
                lateinit var sunrise: String
                @SerializedName("sunset")
                @Expose
                lateinit var sunset: String
                @SerializedName("moonrise")
                @Expose
                lateinit var moonrise: String
                @SerializedName("moonset")
                @Expose
                lateinit var moonset: String

            }
        }
    }
}